scripts/dev/compare_output_reports.py

```diff
@@ -0,0 +1,301 @@
+    return manual_html_mapping_dict
+
+
+def parse_html(out_dir):
+    # Lazy load bs4 as its not in the standard library
+    try:
+        from bs4 import BeautifulSoup
+    except ModuleNotFoundError:
+        print("bs4 (BeautifulSoup) is not installed, not checking HTML")
+        return None
+
+    with open(out_dir / "eplustbl.htm", "rb") as f:
+        soup = BeautifulSoup(f.read(), features="lxml")
+
+    manual_html_mapping_dict = find_manual_mapping_html()
+    html_report_to_tables_list = []
+    for table in soup.find_all("table"):
+        table_name = table.find_previous("b").text.strip()
+        for previous_p in table.find_all_previous("p"):
+            if "Report" in previous_p.text:
+                report_name = previous_p.find("b").text.strip()
+                break
+
+        html_report_to_tables_list.append((report_name, table_name))
+
+    html_report_to_tables_list_without_spaces = []
+
+    for report_name, table_name in html_report_to_tables_list:
+        report_name = report_name.replace(" ", "")
+        table_name = table_name.replace(" ", "")
+        if report_name in manual_html_mapping_dict:
+            # print(f"Override for {report_name}")
+            report_name = manual_html_mapping_dict[report_name]
+
+        html_report_to_tables_list_without_spaces.append(
+            (report_name, table_name)
+        )
+
+    return set(html_report_to_tables_list_without_spaces)
+
+
+def compare_sets(
+    lhs: set, lhs_name: str, rhs: set, rhs_name: str, type_str: str
+) -> bool:
+
+    success = True
+
+    extra_lhs = lhs - rhs
+    if extra_lhs:
+        print(f"{type_str} that are in the {lhs_name} but not in {rhs_name}:")
+        print(extra_lhs)
+        success = False
+
+    extra_rhs = rhs - lhs
+    if extra_rhs:
+        print(f"{type_str} that are in the {rhs_name} but not in {lhs_name}:")
+        print(extra_rhs)
+        success = False
+
+    return success
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Compare Output Reports.")
+    parser.add_argument(
+        "out_dir", type=Path, help="Output directory where to find the reports"
+    )
+    args = parser.parse_args()
+
+    out_dir = args.out_dir.resolve()
+    if not (out_dir.exists() and out_dir.is_dir()):
+        raise IOError(f"{out_dir} is not a valid directory")
+
+    (
+        json_report_names_set,
+        json_report_to_tables_set,
+        json_empty_report_to_tables_set,
+    ) = parse_json(out_dir)
+    sql_report_names_set, sql_report_to_tables_set = parse_sql(out_dir)
+
+    success = True
+    success &= compare_sets(
+        lhs=sql_report_names_set,
+        lhs_name="SQL",
+        rhs=json_report_names_set,
+        rhs_name="JSON",
+        type_str="Reports",
+    )
+
+    success &= compare_sets(
+        lhs=sql_report_to_tables_set,
+        lhs_name="SQL",
+        rhs=json_report_to_tables_set,
+        rhs_name="JSON",
+        type_str="Tables",
+    )
```

Comment:
New support script to compare JSON and SQLite outputs (and with HTML too if bs4 is available (BeautifulSoup))

Reply:
This is really great, thanks!